src/write-archive/index.test.ts

```diff
@@ -41,6 +42,76 @@ describe('writeTestResult', () => {
     );
   });
 
+  it('successfully generates test results with mapped source entries', async () => {
+    // @ts-expect-error Jest mock
+    fs.ensureDir.mockReturnValue(true);
+    const oldMockFs = fs.outputFile;
+    const storyJson = {
+      childNodes: [
+        {
+          attributes: {
+            src: '/bogano',
+          },
+        },
+      ],
+    };
+
+    const expectedMappedJson = {
+      childNodes: [
+        {
+          attributes: {
+            src: '/coruscant',
+          },
+        },
+      ],
+    };
+
+    const sourceMapping = new Map<string, string>();
+    sourceMapping.set('/bogano', '/coruscant');
+    await writeTestResult(
+      // the default output directory in playwright
+      { title: 'Toy Story', outputDir: resolve('test-results/toy-story-chromium') } as TestInfo,
+      { home: Buffer.from(JSON.stringify(storyJson)) },
+      {
+        'http://localhost:3000/home': {
+          statusCode: 200,
+          body: Buffer.from(JSON.stringify(storyJson)),
+        },
+      },
+      { viewport: { height: 480, width: 720 } },
+      sourceMapping
+    );
+
+    const expectedBuffer = Buffer.from(JSON.stringify(expectedMappedJson));
+    const oldMock = fs.outputFile;
+
+    // XXX_jwir3: Ugh, this is annoying that I have to do it this way, but I think jest is caching the mocks
+    //            somehow. I always get the wrong result when I use expect(fs.outputFile).toHaveBeenCalledWith().
+    fs.outputFile = (filename: string, data: string | Buffer, callback: fs.NoParamCallback) => {
+      expect(data).toEqual(expectedBuffer);
```

Comment:
This isn't being executed at all. You can see for yourself by changing `expectedBuffer` to anything at all. It will still pass when you run the tests.

Reply:
Yes, I see that now. I assumed it was similar to how the expectations work asynchronously in jest. Ok, back to the drawing board...